src/AbstractNormalizedEventManager.php

```diff
@@ -139,21 +150,38 @@ public function _zRemoveCachedEvent($name)
      *
      * @return $this This instance.
      */
-    protected function _registerCacheClearHandler(EventInterface $event)
+    protected function _registerClearCacheHandler(EventInterface $event)
+    {
+        $priority = static::CACHE_CLEAR_HANDLER_PRIORITY;
+        $callback = $this->_createClearCacheHandler($event);
+
+        $this->_addHook($event->getName(), $callback, $priority);
+
+        return $callback;
+    }
+
+    /**
+     * Creates a cache clear handler.
+     *
+     * @since [*next-version*]
+     *
+     * @param EventInterface $event The event instance to be cleared from cache by the handler.
+     *
+     * @return \callable The created handler.
+     */
+    protected function _createClearCacheHandler(EventInterface $event)
     {
         $me       = $this;
         $priority = static::CACHE_CLEAR_HANDLER_PRIORITY;
 
         $callback = function($value) use ($me, $event, &$callback, $priority) {
             $me->_zRemoveCachedEvent($event->getName());
-            $me->_detach($event, $callback, $priority);
+            $me->detach($event, $callback, $priority);
```

Comment:
See [this comment](https://github.com/Dhii/wp-events/pull/10/files/4c0b8e7ab21ecb2d4eecd234aec6d177727acf6a..a8480673207370a699a71a74643ccbfff7663ae2#r103468882).